tests/queries/0_stateless/02302_column_decl_null_before_defaul_value.sql

```diff
@@ -0,0 +1,51 @@
+select 'create table, column +type +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +DEFAULT +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT DEFAULT 1 NULL) ENGINE=MergeTree() ORDER BY tuple();
```

Comment:
Going back to the column specification in the documentation which I referenced above:

`name1 [type1] [NULL|NOT NULL] [DEFAULT|MATERIALIZED|EPHEMERAL|ALIAS expr1] [compression_codec] [TTL expr1],`

It states that (NOT) NULL must precede DEFAULT. So either the documentation is now wrong or the implementation allows for undocumented syntax ^^

Reply:
See #37087 #35889